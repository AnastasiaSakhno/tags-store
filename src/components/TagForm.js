import React, { Component } from 'react'
import PropTypes from 'prop-types'

class TagForm extends Component {
  static propTypes = {
    onTagSubmit: PropTypes.func.isRequired
  }

  handleSubmit = (e) => {
    e.preventDefault()

    const name = this.tagNameInput.value
    if(!name) {
      return
    }

    this.props.onTagSubmit({ name: name, linkId: this.props.linkId })

    this.tagNameInput.value = ''
  }

  render() {
    return (
      <form className='tag-form' onSubmit={ this.handleSubmit }>
        <input type='text' placeholder='Tag name' ref={ (el) => { this.tagNameInput = el } }/>
        <input type='submit' value='Add tag'/>
      </form>
    )
  }
}

TagForm.propTypes = {
  linkId: PropTypes.string.isRequired
}

export default TagForm
