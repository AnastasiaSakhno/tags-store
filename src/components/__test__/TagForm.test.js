import React from 'react'
import renderer from 'react-test-renderer'
import TagForm from '../TagForm'
import uniqId from '../../utils/uniq-id'

const linkId = uniqId()
const addTag = () => {}

it('renders correctly', () => {
  const tree = renderer
    .create(<TagForm onTagSubmit={ addTag } linkId={ linkId }/>)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
