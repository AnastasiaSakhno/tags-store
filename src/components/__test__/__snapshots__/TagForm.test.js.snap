// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<form
  className="tag-form"
  onSubmit={[Function]}
>
  <input
    placeholder="Tag name"
    type="text"
  />
  <input
    type="submit"
    value="Add tag"
  />
</form>
`;
